ert_storage/endpoints/compute/misfits.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+import pandas as pd
+from uuid import UUID
+from typing import Any, Mapping, Optional, List
+import sqlalchemy as sa
+from fastapi import (
+    APIRouter,
+    Depends,
+)
+from ert_storage.database import Session, get_db
+from ert_storage import database_schema as ds, json_schema as js
+
+router = APIRouter(tags=["misfits"])
+
+
+def _calculate_misfit(obs_value, response_value, obs_std) -> List[float]:
+    difference = response_value - obs_value
+    misfit = (difference / obs_std) ** 2
+    return (misfit * np.sign(difference)).tolist()
+
+
+@router.get(
+    "/compute/univariate_misfits",
+    response_model=Mapping[int, Mapping[str, Any]],
+)
+async def get_response_univariate_misfits(
+    *,
+    db: Session = Depends(get_db),
+    ensemble_id: UUID,
+    response_name: str,
+    realization_index: Optional[int] = None,
+) -> Mapping[int, Mapping[str, Any]]:
+    """
+    Compute univariate misfits for response(s)
+    """
+    ensemble = db.query(ds.Ensemble).filter_by(id=ensemble_id).one()
+    if realization_index is not None:
+        responses = [
+            (
+                db.query(ds.Record)
+                .filter(
+                    sa.and_(
+                        ds.Record.ensemble_pk == ensemble.pk,
+                        ds.Record.name == response_name,
+                        ds.Record.realization_index == realization_index,
+                    )
+                )
+                .one()
+            )
+        ]
+    else:
+        responses = (
+            db.query(ds.Record)
+            .filter(
+                sa.and_(
+                    ds.Record.ensemble_pk == ensemble.pk,
+                    ds.Record.name == response_name,
+                )
+            )
+            .all()
+        )
+    misfits_dict = {}
+    for response in responses:
+        if (
+            response.observations
+            and response.record_type == ds.RecordType.float_vector
+            and response.realization_index is not None
+        ):
+            data_df = pd.DataFrame(response.f64_matrix.content)
+            labels = response.f64_matrix.labels
+            if labels is not None:
+                data_df.columns = labels[0]
+                data_df.index = labels[1]
+            obs = response.observations[0]
+            misfits_dict[response.realization_index] = _calculate_misfit(
+                np.array(obs.values),
+                data_df.loc[obs.x_axis].values.flatten(),
```

Comment:
This line fetches all rows in `data_df` with indices in`obs.x_axis`? Are we sure the order is correct wrt. `obs.values`? (If not and if it's not easy to fix, open an issue for it so we can handle it later)

Reply:
The order is given by the `pandas.DataFrame` (rows and columns), which is created gracefully :) No issue is necessary.